### PR TITLE
Make basic yaml reader/write public.

### PR DESCRIPTION
**Pull request by MikeStall**
Doesn't yet expose full lexer - wrap it in a dictionary converter for convenience.

### CODE REVIEW

File: src/PAModel/PAConvert/Yaml/YamlConverter.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+namespace Microsoft.PowerPlatform
+{
+    /// <summary>
+    /// Convert a dictionary to and from yaml
+    /// </summary>
+    public static class YamlConverter
+    {
+        public static Dictionary<string, string> Read(TextReader reader, string filenameHint = null)
+        {
+            var d = new Dictionary<string, string>(StringComparer.Ordinal);
```
Comment (lesaltzm) [inline src/PAModel/PAConvert/Yaml/YamlConverter.cs:17]: very minor nit: this deserves a better name than `d`

Comment (MikeStall) [reply src/PAModel/PAConvert/Yaml/YamlConverter.cs]: updated

File: src/PAModelTests/PublicSurfaceTests.cs [hunk 2]
```diff
@@ -26,6 +26,7 @@ public void Test()
                 $"{ns}.{nameof(ChecksumMaker)}",
                 $"{ns}.{nameof(ErrorContainer)}",
                 $"{ns}.{nameof(Error)}",
+                $"Microsoft.PowerPlatform.YamlConverter"
```
Comment (lesaltzm) [inline src/PAModelTests/PublicSurfaceTests.cs:29]: nit: $"Microsoft.PowerPlatform.{nameof(YamlConverter)}"

and add using Microsoft.PowerPlatform;

**Review by lesaltzm — APPROVED** [review]
🚢

**Pull request merged**